src/intrinsic/mod.rs

```diff
@@ -307,6 +308,21 @@ impl<'a, 'gcc, 'tcx> IntrinsicCallMethods<'tcx> for Builder<'a, 'gcc, 'tcx> {
                     }
                 }
 
+                sym::black_box => {
+                    // TODO(@Commeownist): I suspect this can be implemented more efficiently via volatile reads and writes.
```

Comment:
That is likely less efficient as an empty asm block won't emit any instructions, while volatile loads and stores do emit instructions as the volatile prevents them from being removed.

Reply:
Fair enough, but I think it depends on specific use case: those optimizations inhibited by the asm block may be well worth the extra load. Anyway, I don't think this matters much, and programmers can reinvent their own versions of `black_box` just fine. 

I wish we had black box embedded in GCC and LLVM.